File: ru.kmorozov.gbd.desktop/src/main/java/ru/kmorozov/gbd/desktop/gui/MainFrame.java
package ru.kmorozov.gbd.desktop.gui;

import javax.swing.*;
import java.awt.*;

/**
 * Created by km on 05.12.2015.
 */
public class MainFrame extends JFrame {

    private static MainFrame INSTANCE;

    public MainFrame() {
        super("wGooBooDo");

        if (null != MainFrame.INSTANCE) return;

        MainFrame.INSTANCE = this;

        this.setDefaultCloseOperation(WindowConstants.EXIT_ON_CLOSE);

        this.setContentPane(new MainBookForm().getMainPanel());
        this.pack();

        Dimension dimension = Toolkit.getDefaultToolkit().getScreenSize();
        int x = (int) ((dimension.getWidth() - (double) this.getWidth()) / 2.0);
        int y = (int) ((dimension.getHeight() - (double) this.getHeight()) / 2.0);
        this.setLocation(x, y);
    }

    public static MainFrame getINSTANCE() {
        return MainFrame.INSTANCE;
    }

    public void setVisible() {
        this.setVisible(true);
    }
}

File: ru.kmorozov.gbd.desktop/src/main/java/ru/kmorozov/gbd/desktop/gui/MainFrame.kt
package ru.kmorozov.gbd.desktop.gui

import javax.swing.*
import java.awt.*

/**
 * Created by km on 05.12.2015.
 */
class MainFrame : JFrame("wGooBooDo") {
    init {
        instance = this

        defaultCloseOperation = WindowConstants.EXIT_ON_CLOSE

        contentPane = MainBookForm().mainPanel
        pack()

        val dimension = Toolkit.getDefaultToolkit().screenSize
        val x = ((dimension.getWidth() - width.toDouble()) / 2.0).toInt()
        val y = ((dimension.getHeight() - height.toDouble()) / 2.0).toInt()
        setLocation(x, y)
    }

    fun setVisible() {
        isVisible = true
    }

    companion object {

        lateinit var instance: MainFrame
    }
}
